### PR TITLE
Quarkus Vertx/sql extension scenario

### PR DESCRIPTION
**Pull request by pjgg**
We have created a flight search engine in order to test Quarkus Reactive SQL extensions. 
So, in this hypothetical scenario you could have a website that allow to your costumer compare flights prices of different airlines based on some business rules.

### Testing scope

Quarkus / Vertx SQL exploratory testing

* Quarkus Vertx Reactive Routes (annotations)
* OpenAPI from annotations
* Reactive SQL / Postgresql / Mysql/ DB2
    * Select / insert statements
    * Transaction statement between several tables 
* Error mapping through routes

### CODE REVIEW

**Comment by rsvoboda** [comment]
@pjgg this is still marked as draft, what are your remaining AIs?

**Comment by pjgg** [comment]
> @pjgg this is still marked as draft, what are your remaining AIs?

Done!

**Comment by rsvoboda** [comment]
@kshpak please do the initial review

**Comment by kshpak** [comment]
@pjgg can you please clean your commit history (squash merge, split if needed, etc.)

File: 303-quarkus-vertx-sql/README.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+| MAD       | Madrid     |
+| BCN       | Barcelona  |
+| LHR       | London     |
+| CDG       | Paris      |
+| FRA       | Frakfurt   |
+| IST       | Istanbul   |
+| AMS       | Amsterdam  |
+| FCO       | Rome       |
+| CPH       | Copenhagen |
+
```
Comment (kshpak) [inline 303-quarkus-vertx-sql/README.md:61]: Where is Brno? :D

File: 303-quarkus-vertx-sql/src/test/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+quarkus.datasource.reactive.url=postgresql://localhost:5432/amadeus
+quarkus.datasource.jdbc.url=jdbc:postgresql://localhost:5432/amadeus
+
+quarkus.datasource.reactive.max-size=5
+quarkus.datasource.reactive.idle-timeout=1
+
+# Flyway
+quarkus.flyway.migrate-at-start=true
+quarkus.flyway.schemas=amadeus
+quarkus.flyway.locations=db/test/migration/postgres,db/test/migration/common
```
Comment (kshpak) [inline 303-quarkus-vertx-sql/src/test/resources/application.properties:23]:
```suggestion
quarkus.flyway.locations=db/test/migration/postgresql,db/test/migration/common
```
?

File: 303-quarkus-vertx-sql/src/test/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# Database
+%postgresql.app.selected.db=postgresql
+quarkus.datasource.db-kind=postgresql
+quarkus.datasource.username=test
+quarkus.datasource.password=test
+quarkus.datasource.reactive.url=postgresql://localhost:5432/amadeus
+quarkus.datasource.jdbc.url=jdbc:postgresql://localhost:5432/amadeus
+
+quarkus.datasource.reactive.max-size=5
+quarkus.datasource.reactive.idle-timeout=1
```
Comment (kshpak) [inline 303-quarkus-vertx-sql/src/test/resources/application.properties:18]: As opposed to src/main it's 10, is it intentional?

Comment (pjgg) [reply 303-quarkus-vertx-sql/src/test/resources/application.properties]: Yes, for test as DBpool connections IDLE it's easier to test (will takes lees time) with 1 sec than 10 sec.

**Review by kshpak — APPROVED** [review]

File: 303-quarkus-vertx-sql/README.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+<img src="https://media4.giphy.com/media/l0MYs84E42CL9I9JC/giphy.gif?cid=ecf05e474m9in4t30rbqkspwiylltm8i34is1jvk8cmzshty&rid=giphy.gif" title="Find your flight!" >
+
+## Requirements
+
+To compile and run this demo you will need:
+
+- JDK 11+
+- Docker
+- Postgresql
```
Comment (rsvoboda) [inline 303-quarkus-vertx-sql/README.md:11]:
```suggestion
- PostgreSQL
```

File: 303-quarkus-vertx-sql/README.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+- Mysql
```
Comment (rsvoboda) [inline 303-quarkus-vertx-sql/README.md:12]:
```suggestion
- MySQL
```

File: 303-quarkus-vertx-sql/README.md (hunk 4 above)
Comment (rsvoboda) [reply 303-quarkus-vertx-sql/README.md:11]: Plus the other places where the same text appears

File: 303-quarkus-vertx-sql/README.md (hunk 5 above)
Comment (rsvoboda) [reply 303-quarkus-vertx-sql/README.md:12]: Plus the other places where the same text appears

File: 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/dbpool/DbPostgresPoolTest.java [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+            LOGGER.info("###################################################: ");
+            Multi.createFrom().range(1, 3)
+                    .concatMap(n -> {
+                        LOGGER.info("Connection #" + at.incrementAndGet());
+                        return postgresql.preparedQuery("SELECT CURRENT_TIMESTAMP")
+                                .execute().onFailure().invoke(error -> {
+                                    LOGGER.info("Error: " + at.get());
+                                    LOGGER.error("Error on query: '" + error.getMessage() + "'");
+                                    latch.countDown();
+                                    Assertions.assertTrue(false, error.getMessage());
```
Comment (rsvoboda) [inline 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/dbpool/DbPostgresPoolTest.java:157]: https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#fail-java.lang.String-

**Review by rsvoboda — COMMENTED** [review]
@pjgg only wording

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**